ush/python/pygfs/task/marine_analysis.py

```diff
@@ -0,0 +1,490 @@
+           directory structure to run the SOCA variational application
+        """
+        logger.info(f"---------------- Setup runtime environement")
+
+        anl_dir = self.task_config.DATA
+
+        # create analysis directories
+        diags = os.path.join(anl_dir, 'diags')            # output dir for soca DA obs space
+        obs_in = os.path.join(anl_dir, 'obs')             # input      "           "
+        anl_out = os.path.join(anl_dir, 'Data')           # output dir for soca DA
+        FileHandler({'mkdir': [diags, obs_in, anl_out]}).sync()
+
+        # stage fix files
+        logger.info(f"Staging SOCA fix files from {self.task_config.SOCA_INPUT_FIX_DIR}")
+        soca_fix_list = parse_j2yaml(self.task_config.SOCA_FIX_YAML_TMPL, self.task_config)
+        FileHandler(soca_fix_list).sync()
+
+        # prepare the MOM6 input.nml
+        mdau.prep_input_nml(self.task_config)
+
+        # stage the soca utility yamls (gridgen, fields and ufo mapping yamls)
+        logger.info(f"Staging SOCA utility yaml files from {self.task_config.PARMsoca}")
+        soca_utility_list = parse_j2yaml(self.task_config.MARINE_UTILITY_YAML_TMPL, self.task_config)
+        FileHandler(soca_utility_list).sync()
+
+    @logit(logger)
+    def _prep_variational_yaml(self: Task) -> None:
+        """Create the yaml configuration to run the SOCA variational application
+        """
+
+        # prepare background list for the pseudo model, check bkg date for consistency
+        mdau.gen_bkg_list(bkg_path='./bkg',
+                          window_begin=self.task_config.MARINE_WINDOW_BEGIN,
+                          yaml_name='bkg_list.yaml')
+
+        # Make a copy of the env config before modifying to avoid breaking something else
+        envconfig_jcb = copy.deepcopy(self.task_config)
+        logger.info(f"---------------- Prepare the yaml configuration")
+        logger.info(f"{envconfig_jcb}")       # Prepare the yaml configuration
+
+        # Add the things to the envconfig in order to template JCB files
+        envconfig_jcb['PARMgfs'] = self.task_config.PARMgfs
+        envconfig_jcb['nmem_ens'] = self.task_config.NMEM_ENS
+        envconfig_jcb['berror_model'] = 'marine_background_error_static_diffusion'
+        if self.task_config.NMEM_ENS > 3:
+            envconfig_jcb['berror_model'] = 'marine_background_error_hybrid_diffusion_diffusion'
+        envconfig_jcb['DATA'] = self.task_config.DATA
+        envconfig_jcb['OPREFIX'] = self.task_config.OPREFIX
+        envconfig_jcb['PDY'] = os.getenv('PDY')
+        envconfig_jcb['cyc'] = os.getenv('cyc')
+        envconfig_jcb['SOCA_NINNER'] = self.task_config.SOCA_NINNER
+        envconfig_jcb['obs_list'] = ['adt_rads_all']
+
+        # Write obs_list_short
+        with open('obs_list_short.yaml', 'w') as file:
+            yaml.dump(parse_obs_list_file(self.task_config.MARINE_OBS_LIST_YAML), file, default_flow_style=False)
```

Comment:
Prefer using the `wxflow` `save_as_yaml` method.  See, e.g. https://github.com/NOAA-EMC/global-workflow/blob/ec634926f3b1ba1984b0735e17b76869659ceabe/ush/python/pygfs/task/atm_analysis.py#L104